cnf-certification-test/observability/suite.go

```diff
@@ -17,14 +17,79 @@
 package observability
 
 import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+
 	"github.com/onsi/ginkgo/v2"
 	"github.com/sirupsen/logrus"
 	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/common"
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/identifiers"
+	"github.com/test-network-function/cnf-certification-test/internal/clientsholder"
+	"github.com/test-network-function/cnf-certification-test/pkg/provider"
+	"github.com/test-network-function/cnf-certification-test/pkg/tnf"
+	v1 "k8s.io/api/core/v1"
 )
 
 //
 // All actual test code belongs below here.  Utilities belong above.
 //
 var _ = ginkgo.Describe(common.ObservabilityTestKey, func() {
-	logrus.Debugf("%s not moved yet to new framework", common.ObservabilityTestKey)
+	var env provider.TestEnvironment
+	ginkgo.BeforeEach(func() {
+		provider.BuildTestEnvironment()
+		env = provider.GetTestEnvironment()
+	})
+
+	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestLoggingIdentifier)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		testContainersLogging(&env)
+	})
 })
+
+//nolint:funlen
+func testContainersLogging(env *provider.TestEnvironment) {
+	if len(env.Containers) == 0 {
+		ginkgo.Skip("No containers to run test, skipping")
+	}
+
+	ocpClient := clientsholder.NewClientsHolder()
+
+	// Iterate through all the CUTs to get their log output. The TC checks that at least
+	// one log line is found.
+	numLogLines := int64(1)
+	badContainers := []string{}
+	for _, cut := range env.Containers {
+		podLogOptions := v1.PodLogOptions{TailLines: &numLogLines, Container: cut.Data.Name}
```

Comment:
can you export all this code between lines [64:89] into a separate function ?

Reply:
Ok, will do.